Composer/packages/server/src/externalContentProvider/externalContentProvider.ts

```diff
@@ -53,4 +54,9 @@ export abstract class ExternalContentProvider<T extends IContentProviderMetadata
    * (Optional) Performs any necessary authentication for the service and returns an access token.
    */
   abstract authenticate?(): Promise<string>;
+
+  /**
+   * get different type of profiles base on provider
```

Comment:
"(Optional) Generates a publish profile based on the provider."

Reply:
Seem like abstract methods must be implemented in child class even it's optional, So I remove the abstract and make it just optional.